packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -97,261 +105,282 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
+    setGroups([]);
+    const groups = await fetchGroups(
+      setGroups,
+      iModelId,
+      mapping.id ?? "",
+      setIsLoading,
+      apiContext,
+    );
+    if (groups) {
+      if (showGroupColor) {
+        await visualizeGroupColors(groups);
+      } else {
+        clearEmphasizedOverriddenElements();
+      }
     }
-    return colour;
-  };
+  }, [
+    apiContext,
+    iModelId,
+    mapping.id,
+    setGroups,
+    showGroupColor,
+    visualizeGroupColors,
+  ]);
 
-  const onSelect = useCallback(
-    async (selectedData: GroupType[] | undefined) => {
+  const resetView = useCallback(async () => {
+    if (groups) {
+      if (showGroupColor) {
+        await visualizeGroupColors(groups);
+      } else {
+        clearOverriddenElements();
+      }
       clearEmphasizedElements();
-      if (selectedData && selectedData.length > 0) {
-        setLoadingQuery(true);
-        let allIds: string[] = [];
-        for (const row of selectedData) {
-          const query = row.query ?? "";
-          if (hilitedElements.current.has(query)) {
-            const hilitedIds = hilitedElements.current.get(query) ?? [];
-            visualizeElements(hilitedIds, stringToColor(row.id ?? ""));
-            allIds = allIds.concat(hilitedIds);
-          } else {
-            try {
-              const ids: string[] = await fetchIdsFromQuery(
-                query,
-                iModelConnection,
-              );
-              if (ids.length === 0) {
-                toaster.warning(`${row.groupName}'s query is valid but produced no results.`);
-              }
-              const hiliteIds = await visualizeElementsById(
-                ids,
-                stringToColor(row.id ?? ""),
-                iModelConnection,
-              );
-              hilitedElements.current.set(query, hiliteIds);
+    }
+  }, [
+    groups,
+    showGroupColor,
+    visualizeGroupColors,
+  ]);
 
-              allIds = allIds.concat(ids);
-            } catch {
-              const index = groups.findIndex((group) => group.id === row.id);
-              setSelectedRows((rowIds) => {
-                const selectedRowIds = { ...rowIds };
-                delete selectedRowIds[index];
-                return selectedRowIds;
-              });
-              toaster.negative(`Could not load ${row.groupName}. Query could not be resolved.`);
+  const showAll = useCallback(async () => {
+    clearHiddenElements();
+    setHiddenGroupsIds([]);
+    await zoomToElements(Array.from(hilitedElements.current.values()).flat());
+  }, [setHiddenGroupsIds]);
 
-            }
-          }
-        }
-        await zoomToElements(allIds);
-        setLoadingQuery(false);
-      }
-    },
-    [iModelConnection, groups],
-  );
+  const hideAll = useCallback(async () => {
+    await hideGroups(groups);
+    setHiddenGroupsIds(
+      groups.filter((g) => g.id).map((g) => (g.id ? g.id : "")),
+    );
+  }, [setHiddenGroupsIds, groups, hideGroups]);
```

Comment:
For hideAll or maybe showAll as well, you should probably wrap it with `setLoadingQuery(true/false)` because sometimes it takes a while. I've also noticed the buttons sometimes become unresponsive. I didn't look too far into it.